### PR TITLE
Update liquidation price estimate spec

### LINKED ISSUE #2075: Update margin spec for estimate margin with the introduction of isolated margin
Now that the isolated margin work has started we need to update margin spec for estimate margin with the introduction of isolated margin:

- [x] update the spec
- [x] Add ACs as required

### PR DESCRIPTION
**Pull request by witgaw**
Update spec to consider isolated margin mode.

Close https://github.com/vegaprotocol/specs/issues/2075

### CODE REVIEW

File: non-protocol-specs/0012-NP-LIPE-liquidation-price-estimate.md [hunk 1]
```diff
@@ -50,6 +50,14 @@ When including orders we sort the orders in the order they will get filled in (d
     - update $S^{\text{current}}$ to equal $S^{\text{order}}$,
   - otherwise return last calculated $S^{\text{liquidation}}$ (assumes other orders will get cancelled and the remaining position will be liquidated).
 
+### Margin mode considerations
+
+Depending on the [margining mode](../protocol/0019-MCAL-margin_calculator.md#margining-modes) selected by the party for the market on which its position is being considered the $\text{collateral available}$ will differ.
+
+Cross-margin mode: $\text{collateral available} = \text{margin account balance} + \text{general account balance}$.
+
+Isolated margin mode: $\text{collateral available} = \text{margin account balance} + \text{order margin account balance}$.
```
Comment (davidsiska-vega) [inline non-protocol-specs/0012-NP-LIPE-liquidation-price-estimate.md]: @TomMcL / @Jiajia-Cui can you please check if this is correct for isolated margin? 

My impression was that to prop up a position in isolated mode it's only the margin account balance.

Comment (Jiajia-Cui) [reply non-protocol-specs/0012-NP-LIPE-liquidation-price-estimate.md]: That is right, so it should be 

isolated margin mode: $\text{collateral available} = \text{margin account balance}

Comment (TomMcL) [reply non-protocol-specs/0012-NP-LIPE-liquidation-price-estimate.md]: Agreed, order margin account shouldn't be touched until the actual positions have already been liquidated/will be

Comment (witgaw) [reply non-protocol-specs/0012-NP-LIPE-liquidation-price-estimate.md]: Thanks, amended now.

File: non-protocol-specs/0012-NP-LIPE-liquidation-price-estimate.md [hunk 2]
```diff
@@ -58,3 +66,4 @@ When including orders we sort the orders in the order they will get filled in (d
...
 1. An estimate is obtained for a short position with multiple limit sell order with the absolute value of the total remaining size of the orders less than the open volume. The estimated liquidation price with sell orders is lower than that for the open volume only. As the limit orders get filled the estimated liquidation price for the (updated) open volume converges to the estimate originally obtained with open sell orders. As the price keeps moving in small increments the liquidation happens within the originally estimated range (with sell orders) (<a name="0012-NP-LIPE-005" href="#0012-NP-LIPE-005">0012-NP-LIPE-005</a>)
 1. There's no difference in the estimate for an open volume and that with `0` open volume and market order of the same size. (<a name="0012-NP-LIPE-006" href="#0012-NP-LIPE-006">0012-NP-LIPE-006</a>)
+1. When margining mode gets successfully changed and party has non-zero general account balance afterwards than its liquidation price estimate for all cases (position only, with buy orders, with sell orders) moves closer to the current mark price (compared to cross-margin figure). (<a name="0012-NP-LIPE-007" href="#0012-NP-LIPE-007">0012-NP-LIPE-007</a>)
```
Comment (gordsport) [inline non-protocol-specs/0012-NP-LIPE-liquidation-price-estimate.md]: Can this new AC please also be added to the features.json file?

Comment (gordsport) [reply non-protocol-specs/0012-NP-LIPE-liquidation-price-estimate.md]: https://github.com/vegaprotocol/specs/pull/2079/commits/b9eaabb05f853b931649ae0942d1feabca8be25e

**Review by davidsiska-vega — APPROVED** [review]

**Pull request merged**